gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/AbstractChannelizer.java

```diff
@@ -147,7 +149,12 @@ public void init(final ServerGremlinExecutor<EventLoopGroup> serverGremlinExecut
     public void initChannel(final SocketChannel ch) throws Exception {
         final ChannelPipeline pipeline = ch.pipeline();
 
-        if (sslContext.isPresent()) pipeline.addLast(PIPELINE_SSL, sslContext.get().newHandler(ch.alloc()));
+        sslContext.ifPresent(sslContext -> pipeline.addLast(PIPELINE_SSL, sslContext.newHandler(ch.alloc())));
+
+        // checks for no activity on a channel and triggers an event that is consumed by the OpSelectorHandler
+        // and either closes the connection or sends a ping to see if the client is still alive
+        if (supportsIdleMonitor())
+            pipeline.addLast(new IdleStateHandler((int) (settings.idleReadLimit / 1000), (int) (settings.idleWriteLimit / 1000),0));
```

Comment:
Is this setting in millis or seconds?

Reply:
The constructor for `IdleStateHandler` is seconds but all Gremlin Server time settings are milliseconds so i had to convert: https://netty.io/4.0/api/io/netty/handler/timeout/IdleStateHandler.html#IdleStateHandler-int-int-int-

Reply:
Then I think we should use seconds as a time unit for this setting :)

Reply:
I think we should keep it millis for consistency across configuration options.  There could be a note stating it has only second-resolution.

Reply:
yeah - i have that same feeling @robertdale - i'm considering some validation or something there in addition to documentation fixes.

Reply:
Adding some general logging that explains the conversion:

```text
[INFO] GremlinServer - idleConnectionTimeout was set to 500 which resolves to 0 seconds when configuring this value - this feature will be disabled
[INFO] GremlinServer - keepAliveInterval was set to 0 which resolves to 0 seconds when configuring this value - this feature will be disabled
```